### PR TITLE
Shared tsconfig

### PR DESCRIPTION
**Pull request by clemsos**
# Description

Typescript config as a package

WIP -- ~~needs to be rebased when #7575 will be merged~~ rebased

# Issues

Relates to #7427 

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by clemsos** [comment]
so finally, what happened here

- create a shared package for tsconfig
- upgrade to latest yarn (3.1.x)
- split paywall nested dep into `paywall-app` and `packages/paywall`
- moved unlock-js to `packages/unlock-js`
- had to adapt scripts to make CI happy

The good is: we are starting to have a much more consistent monorepo. For instance adding a new workflow to publish packages should be very easy (w lerna)

The bad is: we broke paywall app deployment. (Need a proper way to import paywall.js)

The ugly: none ofc, except maybe we are now using yarn canary which may or may not blow up at some point. Living on the edge!

**Comment by julien51** [comment]
> Creating docker_paywall-app_run ... done^@^@Failed to deploy to Netlify because we're missing SITE_ID and/or AUTH_TOKEN

Hum. Did this change? It should be set from the CI env vars

**Comment by clemsos** [comment]
> > Creating docker_paywall-app_run ... done^@^@failed to deploy to Netlify because we're missing SITE_ID and/or AUTH_TOKEN
> 
> Hum. Did this change? It should be set from the CI env vars

yes `PAYWALL` should be `PAYWALL_APP` ...

**Comment by julien51** [comment]
Just updated all the variables.

File: packages/paywall/src/networkConfigs.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import networks from '@unlock-protocol/networks'
-import { NetworkConfigs } from '../unlockTypes'
+import { NetworkConfigs } from '@unlock-protocol/types'
```
Comment (julien51) [inline packages/paywall/src/networkConfigs.ts:2]: That's really cool BTW! I really wanted to have repo-wide types for a looong time!

File: paywall-app/next.config.js [hunk 2]
```diff
@@ -2,17 +2,15 @@
 const fs = require('fs')
 const { join } = require('path')
 const { promisify } = require('util')
-const withCSS = require('@zeit/next-css')
```
Comment (julien51) [inline paywall-app/next.config.js:5]: yay!

**Review by julien51 — APPROVED** [review]
💯

File: packages/paywall/src/networkConfigs.ts (hunk 1 above)
Comment (clemsos) [reply packages/paywall/src/networkConfigs.ts:2]: Yes! Next step is to publish as '@types/unlock-protocol' ;)

**Pull request merged**